test/e2e/prometheus_test.go

```diff
@@ -92,3 +97,180 @@ func TestPrometheusAlertmanagerAntiAffinity(t *testing.T) {
+	ctx := context.Background()
+
+	targetDeployment := `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: instrumented-sample-app
+  namespace: openshift-monitoring
+  labels:
+    group: test
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      group: test
+  template:
+    metadata:
+      labels:
+        group: test
+    spec:
+      containers:
+      - name: example-app
+        args:
+        - --cert-path=/etc/certs
+        image: quay.io/coreos/instrumented-sample-app:0.2.0-bearer-mtls-1
+        imagePullPolicy: IfNotPresent
+        ports:
+        - name: web
+          containerPort: 8080
+        - name: mtls
+          containerPort: 8081
+        volumeMounts:
+	- mountPath: /etc/certs
+	  name: certs
+        - mountPath: /etc/certs/tls
+          name: rw-target-tls
+        - mountPath: /etc/certs/ca
+          name: serving-certs-ca-bundle
+      volumes:
+      - name: rw-target-tls
+        secret:
+          secretName: rw-target-tls
+      - name: serving-certs-ca-bundle
+        configMap:
+	  name: serving-certs-ca-bundle
+      - name: certs
+        emptyDir: {}
+      initContainers:
+      - command:
+        - /bin/sh
+        - -c
+        - 'cp /etc/certs/tls/tls.key /etc/certs/key.pem; cat /etc/certs/tls/tls.crt > /etc/certs/cert.pem; cat /etc/certs/ca/service-ca.crt >> /etc/certs/cert.pem'
+	name: crypt-init
+	image: busybox
+	volumeMounts:
+	- mountPath: /etc/certs
+	  name: certs
+        - mountPath: /etc/certs/tls
+          name: rw-target-tls
+        - mountPath: /etc/certs/ca
+          name: serving-certs-ca-bundle
+`
+	deployment := &appsv1.Deployment{}
+	if err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(targetDeployment)), 100).Decode(&deployment); err != nil {
+		t.Fatal(err)
+	}
+
+	// create service with annotation for service-ca operator
+	name := "test"
+	svc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"group": name,
+			},
+			Namespace: f.Ns,
+			Annotations: map[string]string{
+				"service.beta.openshift.io/serving-cert-secret-name": "rw-target-tls",
+			},
+		},
+		Spec: v1.ServiceSpec{
+			Type: v1.ServiceTypeLoadBalancer,
+			Ports: []v1.ServicePort{
+				{
+					Name: "web",
+					Port: 8080,
+				},
+				{
+					Name: "mtls",
+					Port: 8081,
+				},
+			},
+			Selector: map[string]string{
+				"group": name,
+			},
+		},
+	}
+
+	if err := f.OperatorClient.CreateOrUpdateService(ctx, svc); err != nil {
+		t.Fatal(err)
+	}
+	// deploy remote write target
+	if err := f.OperatorClient.CreateOrUpdateDeployment(ctx, deployment); err != nil {
+		t.Fatal(err)
+	}
+	svc, err := f.KubeClient.CoreV1().Services(f.Ns).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, scenario := range []struct {
+		name  string
+		proto string
+	}{
+		// check remote write logs
+		{
+			name:  "assert remote write to http works",
+			proto: "http",
+		},
+	} {
```

Comment:
Since you wrote this test with the intent of having multiple test cases in the future, it would be great to split them into sub-tests via `t.Run()` and differentiate them via their names.